get_nyt_articles.py

```diff
@@ -0,0 +1,88 @@
+'''
+Reads in articles from the New York Times API and saves
+them to a cache.
+'''
+
+import time
+import os
+import datetime
+from nytimesarticle import articleAPI
+import pandas as pd
+# News archive api
+import requests
+
+from dateutil.rrule import rrule, MONTHLY
+NYT_KEY = open('nyt_key.txt').read().strip()
+api = articleAPI(NYT_KEY)
+
+def parse_articles(articles):
+    '''
+    This function takes in a response to the NYT api and parses
+    the articles into a list of dictionaries
+    '''
+    news = []
+    for i in articles['response']['docs']:
+        if 'abstract' not in i.keys():
+            continue
+        if 'headline' not in i.keys():
+            continue
+        if 'news_desk' not in i.keys():
+            continue
+        if 'pub_date' not in i.keys():
+            continue
+        if 'snippet' not in i.keys():
+            continue
+        dic = {}
+        dic['id'] = i['_id']
+        if i.get('abstract', 'EMPTY') is not None:
+            dic['abstract'] = i.get('abstract', 'EMPTY').encode("utf8")
+        dic['headline'] = i['headline']['main'].encode("utf8")
+        dic['desk'] = i.get('news_desk', 'EMPTY')
+        dic['date'] = i['pub_date'][0:10] # cutting time of day.
```

Comment:
Do some length checking here

Reply:
I've added a check for the length of the "pub_date" field in the response.